modules/aws/rds/main.tf

```diff
@@ -46,47 +46,69 @@ module "db" {
-
-  instance_class                        = var.instance_class
-  username                              = var.username
-  password                              = var.password
-
-  snapshot_identifier                   = var.snapshot_identifier
-
-  subnet_ids = var.vpc.database_subnets
-  vpc_security_group_ids = [module.security_group.security_group_id]
-
-  apply_immediately                     = true
-  skip_final_snapshot                   = false
-  auto_minor_version_upgrade            = true
-  backup_retention_period               = 7
-  backup_window                         = "02:21-02:51"
-  copy_tags_to_snapshot                 = true
-  delete_automated_backups              = false
-  deletion_protection                   = false
-  iam_database_authentication_enabled   = false
-  license_model                         = var.engine == "postgres" ? "postgresql-license" : ""
-  maintenance_window                    = "tue:04:29-tue:04:59"
   
+  /*Storage*/
+  storage_encrypted                     = var.storage_encrypted
+  storage_type                          = var.storage_type
+  kms_key_id                            = var.kms_key_id
+  allocated_storage                     = var.storage
   iops                                  = var.iops
   max_allocated_storage                 = var.max_allocated_storage
+  /* DB */
+  instance_class                        = var.instance_class
+  username                              = var.username
+  password                              = var.password
   multi_az                              = var.multi_az
-
   port                                  = local.port
   publicly_accessible                   = var.visibility == "public" ? "true" : "false"
-  storage_encrypted                     = var.storage_encrypted
-  storage_type                          = var.storage_type
-
-  performance_insights_enabled          = var.performance_insights_enabled
-  performance_insights_retention_period = 7
+  
+  snapshot_identifier                   = var.snapshot_identifier
+  apply_immediately                     = var.apply_immediately
+  skip_final_snapshot                   = var.skip_final_snapshot
+  auto_minor_version_upgrade            = var.auto_minor_version_upgrade
+  backup_window                         = var.backup_window
+  backup_retention_period               = var.backup_retention_period
+  maintenance_window                    = var.maintenance_window
+  copy_tags_to_snapshot                 = var.copy_tags_to_snapshot
+  delete_automated_backups              = var.delete_automated_backups
+  deletion_protection                   = var.deletion_protection
+  iam_database_authentication_enabled   = var.iam_database_authentication_enabled
+  license_model                         = var.engine == "postgres" ? "postgresql-license" : ""
+  
+  /* Monitoring */
   enabled_cloudwatch_logs_exports = var.engine == "postgres" ? ["postgresql", "upgrade"] :  ["general"]
```

Comment:
we should enhance this for mysql/mariadb to include rollback, slowquery, etc.

Reply:
I kept that supported values with me. was just waiting for your confirmation on this. 

Below are supported for rds. let me know what all we should add along with general . 

Valid values (depending on engine): alert, audit, error, general, listener, slowquery, trace, postgresql (PostgreSQL), upgrade (PostgreSQL)."

If you think we can take as parameter, rather than hardcoding i've already created a variable so we can simply change 
like below.
enabled_cloudwatch_logs_exports = var.enabled_cloudwatch_logs_exports

values can be passed from terragrunt.

Reply:
MariaDB, MySQL supports the following logs
  1. slowquery
  2. audit 
  3. general
  4. error
  
The error log is enabled by default. To enable logs and publish logs to cloudwatch, As said in the documentation, it requires CustomPrameter Group, Custom Option Group. 

Ref: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.Concepts.MariaDB.html#USER_LogAccess.MariaDB.PublishtoCloudWatchLogs

With that said, The necessary inputs is handled in terragrunt file.